### PR TITLE
Add mieru plugin

### PR DESCRIPTION
**Pull request by enfein**
A testing NekoBoxForAndroid build can be found at https://github.com/enfein/NekoBoxForAndroid/actions/runs/7177235318

This patch requires installation of mieru plugin, which can be found at https://github.com/MatsuriDayo/plugins/pull/49

### CODE REVIEW

**Comment by enfein** [comment]
@purofle

Can you help to review this patch? I built APKs with this patch, installed to my phone, and verified it is working.

Thank you

APK download: https://github.com/enfein/NekoBoxForAndroid/actions/runs/7403943262 and https://github.com/enfein/NekoBoxPlugins/actions/runs/7403962650

**Review by purofle — CHANGES_REQUESTED** [review]
在 `io.nekohasekai.sagernet.fmt.PluginEntry` 里添加下 Mieru，不然下载会炸

File: app/src/main/java/io/nekohasekai/sagernet/ui/profile/MieruSettingsActivity.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/******************************************************************************
+ *                                                                            *
+ * Copyright (C) 2021 by nekohasekai <contact-sagernet@sekai.icu>             *
```
[the remark this answers is not in the document]
Comment (enfein) [reply app/src/main/java/io/nekohasekai/sagernet/ui/profile/MieruSettingsActivity.kt:3]: 这些有 copyright 的文件是从 SagerNet 项目复制过来的，例如这个文件是从 https://github.com/SagerNet/SagerNet/blob/dev/app/src/main/java/io/nekohasekai/sagernet/ui/profile/MieruSettingsActivity.kt 复制过来的。所以这些 copyright 不能删除。

File: app/src/main/java/io/nekohasekai/sagernet/fmt/mieru/MieruFmt.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                                put("port", finalPort)
+                                put("protocol", protocol)
+                            })
+                        })
+                    })
+                })
+            })
+        })
+    }.toStringPretty()
+}
```
[the remark this answers is not in the document]
Comment (enfein) [reply app/src/main/java/io/nekohasekai/sagernet/fmt/mieru/MieruFmt.kt:53]: 新的 commit 里面优化了嵌套的层数。

**Comment by enfein** [comment]
> 在 `io.nekohasekai.sagernet.fmt.PluginEntry` 里添加下 Mieru，不然下载插件会炸

在新的 commit 里面解决了这个问题。

**Review by purofle — APPROVED** [review]

**Review by alipornew1 — APPROVED** [review]

**Pull request merged**